ffi/src/scan.rs

```diff
@@ -398,5 +407,12 @@ pub unsafe extern "C" fn visit_scan_data(
         callback,
     };
     // TODO: return ExternResult to caller instead of panicking?
```

Comment:
seems pretty easy to address this todo? maybe we can go ahead and do that

Reply:
That will require changes in the `c` code, which I was trying to avoid mostly here. I added a reminder for myself [here](https://github.com/delta-io/delta-kernel-rs/pull/614/commits/d23c4d0ccdfd732f46b47eef58c510ab743a97b8#r1929426354) for when I do the final fixup that requires lots of C changes